src/wp-includes/rest-api.php

```diff
@@ -3389,3 +3389,39 @@ static function ( $status, $error_data ) {
 
 	return new WP_REST_Response( $data, $status );
 }
+
+/**
+ * Checks whether a REST API endpoint request is currently being handled.
+ *
+ * This may be a standalone REST API request, or an internal request dispatched from within a regular page load.
+ *
+ * @since 6.5.0
+ *
+ * @global WP_REST_Server $wp_rest_server REST server instance.
+ *
+ * @return bool True if a REST endpoint request is currently being handled, false otherwise.
+ */
+function wp_is_rest_endpoint(): bool {
+	/* @var WP_REST_Server $wp_rest_server */
+	global $wp_rest_server;
+
+	// Check whether this is a standalone REST request.
+	$is_rest_endpoint = wp_is_rest_request();
+	if ( ! $is_rest_endpoint ) {
+		// Otherwise, check whether an internal REST request is currently being handled.
+		$is_rest_endpoint = isset( $wp_rest_server )
+			&& method_exists( $wp_rest_server, 'is_dispatching' )
```

Comment:
I don't think we want the `method_exists` check. Yes we allow filtering the `rest_get_server()` class, but I think the expectation that you can provide a different class that doesn't subclass `WP_REST_Server` at all is not something we should encourage.